desc/optimize/_constraint_wrappers.py

```diff
@@ -0,0 +1,629 @@
+"""Wrappers for doing STELLOPT/SIMSOPT like optimization."""
```

Comment:
our orthogonal projection, do stellopt/simsopt do that?

Reply:
Eh, sort of. In their case its just removing degrees of freedom, since VMEC bc gives a diagonal A matrix.